apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/file/InterfacePossibleTypeEnumBuilder.kt

```diff
@@ -0,0 +1,38 @@
+package com.apollographql.apollo3.compiler.codegen.kotlin.file
+
+import com.apollographql.apollo3.compiler.codegen.Identifier
+import com.apollographql.apollo3.compiler.codegen.kotlin.CgFile
+import com.apollographql.apollo3.compiler.codegen.kotlin.CgFileBuilder
+import com.apollographql.apollo3.compiler.codegen.kotlin.KotlinContext
+import com.apollographql.apollo3.compiler.ir.IrInterface
+import com.squareup.kotlinpoet.ClassName
+
+internal class InterfacePossibleTypeEnumBuilder(
+    private val context: KotlinContext,
+    private val iface: IrInterface
+) : CgFileBuilder {
+  private val layout = context.layout
+  private val packageName = layout.typePackageName()
+
+  private val simpleName = possibleTypesEnumTypeName(iface.name)
+  private val compiledTypeName = "${layout.compiledTypeName(iface.name)}${Identifier.PossibleTypes}"
+
+  override fun prepare() {
+    context.resolver.registerMapType(simpleName, ClassName(packageName, compiledTypeName))
```

Comment:
As discussed in https://github.com/apollographql/apollo-kotlin/issues/4346#issuecomment-1233231498, you can either not register anything or create your own `registerPossibleTypes` here